app/src/main/java/bankingapp/LogIn.java

```diff
@@ -13,58 +14,7 @@ public class LogIn {
 
     private String accountNumber;
     private String pinNumber;
-    private Scanner scanner;
-
-    public LogIn() {
-        scanner = new Scanner(System.in);
-        setAccountNumber();
-        setPinNumber();
-    }
 
-    public LogIn(Connection connection) throws SQLException{
-        this();
-        checkAccount(connection, getAccountNumber(), getPinNum());
-    }
-
-    /**
-     * Setter method to input the Account Number
-     */
-    public void setAccountNumber() {
-        String accountNumber;
-        do {
-            System.out.println("\nEnter your account number:");
-            accountNumber = scanner.next();
-        } while (accountNumber.equals(null) || accountNumber.trim().isEmpty());
-        this.accountNumber = accountNumber.trim();
-    }
-
-    /**
-     * Getter method to get the user input of the Account Number
-     * @return accountNumber
-     */
-    public String getAccountNumber() {
-        return accountNumber;
-    }
-
-    /**
-     * Setter method to input the PIN Number
-     */
-    public void setPinNumber() {
-        String pinNumber;
-        do {
-            System.out.println("Enter your PIN:");
-            pinNumber = scanner.next();
-        } while (pinNumber.equals(null) || pinNumber.trim().isEmpty());
-        this.pinNumber = pinNumber.trim();
-    }
-
-    /**
-     * Getter method to get the user inputted PIN number
-     * @return pinNumber
-     */
-    public String getPinNum() {
-        return pinNumber;
-    }
```

Comment:
Yes! Removing UI logic from setters is a good thing. We want setters to be predictable, and not interact with the user or filesystem or what have you. The reason is that we use them all the time, all over the place in many contexts, and this code was only valid in one specific context.

For example, if you use the Sprint Framework to build a web app, it will use accessor methods internally to do all sorts of magic behind the scenes, and Sprint will expect to call a setter whenever it wants to.